single_cell/qc.py

```diff
@@ -0,0 +1,222 @@
+import os
+import re
+import pypeliner.managed as mgd
+from single_cell.workflows import align
+import pypeliner
+import sys
+from single_cell.utils import inpututils
+from single_cell.workflows.qc import tasks
+
+def qc_workflow(args):
+    data = inpututils.load_qc_input(args["input_yaml"])
+    config = inpututils.load_config(args)
+    config = config["qc"]
+
+    out_dir = args["out_dir"]
+
+    patients = data.keys()
+
+    #inputs
+    sampledata = {patient: data[patient] for patient in patients}
+
+    #outputs
+    indelreports = {patient: os.path.join(out_dir, patient, "indelreport.html") 
+        for patient in patients
+    }
+    mutationreports = {patient: os.path.join(out_dir, patient, "mutationreport.html") 
+        for patient in patients
+    }
+    grouplevelmafs = {patient: os.path.join(out_dir, patient, "grouplevelmaf.maf") 
+        for patient in patients
+    }
+    grouplevel_high_impact_mafs = {patient: os.path.join(out_dir, patient, "grouplevel_high_impact_maf.maf") 
+        for patient in patients
+    }
+    grouplevel_high_impact_merged_snvs = {patient: os.path.join(out_dir, patient, 
+        "grouplevel_high_impact_merged_snvs.csv") for patient in patients
+    }
+    grouplevel_snvs = {patient: os.path.join(out_dir, patient, "grouplevel_snvs.csv") 
+        for patient in patients
+    }
+    patient_names = {patient: patient for patient in patients}
+
+    workflow = pypeliner.workflow.Workflow(
+        ctx={'docker_image': config['docker']['single_cell_pipeline']}
+    )
+
+    workflow.setobj(
+        obj=mgd.OutputChunks('patient'),
+        value=list(patients),
+    )
+
+    workflow.subworkflow(
+        name='patient_level_qc_workflow',
+        func="single_cell.qc.patient_level_qc_workflow",
+        axes=('patient',),
+        args=(
+            mgd.InputFile('patient', 'patient', fnames=patient_names),
+            data,
```

Comment:
what's data??

Reply:
data = inpututils.load_qc_input(args["input_yaml"])

then inside the function,
    sampledata = data[patient]

Reply:
right, so this contains file paths. 


you need to wrap the filepaths as managed files.

Reply:
it contains sample-specific paths. its passed in where i am working at the patient level. How do i do that?